fccodegen/_printer.py

```diff
@@ -0,0 +1,400 @@
+#
+#   a. Redistributions of source code must retain the above copyright notice,
+#      this list of conditions and the following disclaimer.
+#   b. Redistributions in binary form must reproduce the above copyright
+#      notice, this list of conditions and the following disclaimer in the
+#      documentation and/or other materials provided with the distribution.
+#   c. Neither the name of SymPy nor the names of its contributors
+#      may be used to endorse or promote products derived from this software
+#      without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE.
+#
+import sympy
+import sympy.printing
+
+from sympy.printing.precedence import precedence
+
+#from sympy.core import S, Rational, Pow, Basic, Mul
+#from .printer import Printer
+#from sympy.printing.precedence import PRECEDENCE
+#import mpmath.libmp as mlib
+#from mpmath.libmp import prec_to_dps
+#from sympy.utilities import default_sort_key
+
+# List of functions handled by python's `math` module
+_math_functions = {
+    'acos',
+    'acosh',
+    'asin',
+    'asinh',
+    'atan',
+    'atan2',
+    'atanh',
+    'ceil',     # called ceiling in sympy
+    'cos',
+    'cosh',
+    'exp',
+    'expm1',
+    'factorial',
+    'floor',
+    'log',
+    'log10',
+    'log1p',
+    'log2',
+    'sin',
+    'sinh',
+    'tan',
+    'tanh',
+}
+
+
+class WebLabPrinter(sympy.printing.printer.Printer):
+    """
+    Converts Sympy expressions to strings for use in Web Lab Cython model code.
+
+    To use, create a :class:`WebLabPrinter` instance, and call its method
+    :meth:`doprint()` with a Sympy expression argument.
+
+    Arguments:
+
+    ``symbol_function``
+        A function that converts symbols to strings (variable names).
+    ``derivative_function``
+        A function that converts derivatives to strings.
+
+    """
+
+    def __init__(self, symbol_function=None, derivative_function=None):
+        super(WebLabPrinter, self).__init__(None)
+
+        # Prefix for functions
+        self._prefix = 'math.'
+
+        # Symbol and derivative handling (default)
+        if symbol_function is None:
+            self._symbol_function = lambda x: str(x)
```

Comment:
You should be able to use `str` directly rather than the `lambda`

Reply:
Done!